File: output/gpt_4o/avatar/algo_based_translation/python/java/codeforces_8_B.java

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class codeforces_8_B {

    public static String checkMovements(String recordOfMovements) {
        int x = 0;
        int y = 0;
        Map<Character, int[]> dMovement = new HashMap<>();
        dMovement.put('L', new int[]{0, -1});
        dMovement.put('R', new int[]{0, 1});
        dMovement.put('U', new int[]{1, 0});
        dMovement.put('D', new int[]{-1, 0});

        Map<String, String> d = new HashMap<>();
        d.put(x + "," + y, "");
        boolean bug = false;

        for (char move : recordOfMovements.toCharArray()) {
            int[] temp = new int[]{x, y};
            x += dMovement.get(move)[0];
            y += dMovement.get(move)[1];

            String currentPosition = x + "," + y;
            if (!d.containsKey(currentPosition)) {
                for (char step : dMovement.keySet()) {
                    int newX = x + dMovement.get(step)[0];
                    int newY = y + dMovement.get(step)[1];
                    String neighborPosition = newX + "," + newY;
                    if (!(newX == temp[0] && newY == temp[1]) && d.containsKey(neighborPosition)) {
                        bug = true;
                        break;
                    }
                }
            } else {
                bug = true;
            }
            if (bug) {
                break;
            }
            d.put(currentPosition, "");
        }

        return bug ? "BUG" : "OK";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String recordOfMovements = scanner.nextLine();
        System.out.println(checkMovements(recordOfMovements));
        scanner.close();
    }
}

File: dataset/avatar/Python/Code/codeforces_8_B.py
record_of_movements = input ( )
x , y = 0 , 0
d_movement = { 'L' : ( 0 , - 1 ) , 'R' : ( 0 , 1 ) , 'U' : ( 1 , 0 ) , 'D' : ( - 1 , 0 ) }
d = { ( x , y ) : '' }
bug = False
for move in record_of_movements :
    temp = ( x , y )
    x += d_movement [ move ] [ 0 ]
    y += d_movement [ move ] [ 1 ]
    if ( x , y ) not in d :
        for step in d_movement :
            if ( x + d_movement [ step ] [ 0 ] , y + d_movement [ step ] [ 1 ] ) != temp and ( x + d_movement [ step ] [ 0 ] , y + d_movement [ step ] [ 1 ] ) in d :
                bug = True
                break
    else :
        bug = True
    if bug :
        break
    d [ x , y ] = ''
if bug :
    print ( 'BUG' )
else :
    print ( 'OK' )

